### DISCUSSION TITLE
"Sort / Shuffle by Category" options in Inventions List Editor

### DISCUSSION THREAD

**Discussion by kyphii** [OpenRCT2/OpenRCT2#20865] [Feature requests]
I already have this coded up, but wanted to discuss first, primarily to see if this would be considered better off as a plugin rather than in the main repo.

Currently, scenario creation requires a lot of busywork in the inventions list UI - sorting, reordering, and re-shuffling the uninvented list manually. The proposed addition would eliminate the tedious aspects of this to make scenario creation smoother.

Two new buttons to the inventions list UI are added, which allow the player to sort or shuffle the invention lists by invention category.
- Both buttons preserve the order of inventions within each category; eg. if the swinging ship is before the scrambler in the original list, the reordered list will also have the swinging ship before the scrambler.
- The sort button sorts the list, ordering the categories in the standard order (transport -> gentle -> coasters, etc). The invented (top) list is also sorted - this has no effect on gameplay, but will make finding items in a large list easier.
- The shuffle button scrambles the uninvented (bottom) list semi-randomly, also preserving the invention order within the category.

![shuffly](https://github.com/OpenRCT2/OpenRCT2/assets/8711258/15a637ef-aaca-4070-9f52-fa5d0b56d7f3)

**Comment by MASHzj** [comment]
Having tabs per ride type for the invented list, like in the object selection, would also be handy.

**Reply by kyphii** [reply]
I also tried my hand at a "highlight category" dropdown, but dropdowns have so much boilerplate code that I didn't want to bother.
